src/ArmTemplates/Extractor/Utilities/DataProcessors/Absctraction/IGroupDataProcessor.cs

```diff
@@ -0,0 +1,19 @@
+﻿// --------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+// --------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Groups;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities.DataProcessors.Absctraction
+{
+    public interface IGroupDataProcessor
+    {
+        IDictionary<string, string> OverrideRules { get; set; }
+        void ProcessData(List<GroupTemplateResource> template, ExtractorParameters extractorParameters);
```

Comment:
In `IGroupDataProcessor` there is a `List<GroupTemplateResource>` parameter, and in `IProductDataProcessor` there could be a `List<ProductTemplateResource>` parameter. That is instantly showing me that I can make a Generic interface somewhere.

```
IDataProcessor<IList<TTemplateResource>> where TTemplateResource : ITemplateResource // if I remember type names correctly
```